File: 0001-two-sum/0001-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        var numMap = HashMap<Int, Int>()
        
        for(i in nums.indices) {
            var diff = target - nums[i];
            if(numMap.containsKey(diff))
                return intArrayOf(numMap[diff]!!,i)
            numMap.put(nums[i], i)
        }
        return nums
    }
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    var indexOfHashMap = {};
    for (let i = 0; i < nums?.length; i++) {
        const diff = target - nums[i];
        if (diff in indexOfHashMap)
            return [i, indexOfHashMap[diff]];
        indexOfHashMap[nums[i]] = i;
    }
};